### PR TITLE
MAYA-105944 - MayaUSD plugin version in plugin manager doesn't match …

### PR DESCRIPTION
**Pull request by seando-adsk**
#### MAYA-105944 - MayaUSD plugin version in plugin manager doesn't match the release version
* Set the plugin version from the cmake variable 'MAYAUSD_VERSION'

### CODE REVIEW

File: plugin/adsk/plugin/CMakeLists.txt [hunk 1]
```diff
@@ -41,6 +41,13 @@ target_compile_definitions(${TARGET_NAME}
         $<$<BOOL:${UFE_FOUND}>:WANT_UFE_BUILD>
 )
 
+if(DEFINED MAYAUSD_VERSION)
+    target_compile_definitions(${TARGET_NAME}
+        PRIVATE
+            MAYAUSD_VERSION=${MAYAUSD_VERSION}
+    )
+endif()
```
Comment (seando-adsk) [inline plugin/adsk/plugin/CMakeLists.txt:49]: This adds `-DMAYAUSD_VERSION=0.3.0` to the compile line.

File: plugin/adsk/plugin/plugin.cpp [hunk 2]
```diff
@@ -53,6 +53,13 @@
 #include <mayaUsd/ufe/Global.h>
 #endif
 
+#if defined(MAYAUSD_VERSION)
+#define STRINGIFY(x) #x
+#define TOSTRING(x) STRINGIFY(x)
+#else
+#error "MAYAUSD_VERSION is not defined"
+#endif
```
Comment (seando-adsk) [inline plugin/adsk/plugin/plugin.cpp:61]: Macro trick to stringify a define. I cause compiler error if the MAYAUSD_VERSION is not defined.

File: plugin/adsk/plugin/plugin.cpp [hunk 3]
```diff
@@ -78,7 +85,7 @@ MAYAUSD_PLUGIN_PUBLIC
 MStatus initializePlugin(MObject obj)
 {
     MStatus status;
-    MFnPlugin plugin(obj, "Autodesk", "1.0", "Any");
+    MFnPlugin plugin(obj, "Autodesk", TOSTRING(MAYAUSD_VERSION), "Any");
```
Comment (seando-adsk) [inline plugin/adsk/plugin/plugin.cpp:88]: This will be replaced with `"0.3.0"`

Comment (HamedSabri-adsk) [reply plugin/adsk/plugin/plugin.cpp:88]: @seando-adsk  Nice, makes sense. Thanks.

**Pull request merged**